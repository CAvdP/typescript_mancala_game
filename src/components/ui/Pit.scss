.pit {
    transition: all .2s ease-in-out;
    background-color: transparent;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 150px;
    height: 150px;
    font-size: 4rem;
    border-radius: 50%;
    margin: 20px;

    &.active {
        cursor: pointer;
        color: #fff;


        /* outer right cyan */
        &:hover {
            transform: scale(1.1);
        }
    }

}