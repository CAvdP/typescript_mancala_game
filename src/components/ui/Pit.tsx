import React from 'react';
import './Pit.scss';

interface Props {
id: number;
weight: number; // how many seeds in a pit
onClick: (id:number)=>void;
className: string;
}

function Pit({id,weight, onClick, className}: Props){
    let disabled: boolean = weight === 0

    return (
        <button disabled={disabled} className={`pit ${disabled?'disabled':'active ' + className}`} onClick={()=>onClick(id)}>{weight}</button>
    )
}

export default Pit