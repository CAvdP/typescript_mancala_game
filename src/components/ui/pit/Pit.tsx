import React from 'react';
import './styling/Pit.scss';

interface Props {
id: number;
weight: number; // how many seeds in a pit
enabled: boolean;
onClick: (id:number)=>void;
className: string;
}

function Pit({id,weight,enabled, onClick, className}: Props){
    let disabled: boolean = weight === 0 || enabled === false

    return (
        <button disabled={disabled} className={`pit ${disabled?'disabled':'active ' + className}`} onClick={()=>onClick(id)}>{weight}</button>
    )
}

export default Pit