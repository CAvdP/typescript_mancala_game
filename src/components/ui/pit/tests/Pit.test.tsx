import React from 'react'
import renderer from 'react-test-renderer'
import Pit from '../Pit';



describe("Pit Component", () => {
    it('matches the saved snapshot', () => {
        const actualSnapshot = renderer
          .create(<Pit id={0} weight={0} enabled={false} onClick={ (id: number)=>{}}  className={'neon_pink'} />)
          .toJSON()
          
        expect(actualSnapshot).toMatchSnapshot()
      });
});