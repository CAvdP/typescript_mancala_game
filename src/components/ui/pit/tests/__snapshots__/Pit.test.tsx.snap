// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pit Component matches the saved snapshot 1`] = `
<button
  className="pit disabled"
  disabled={true}
  onClick={[Function]}
>
  0
</button>
`;
